.github/workflows/continuous-integration-dotnet.yml

```diff
@@ -0,0 +1,84 @@
+name: Continuous Integration - .NET
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - services/**
+  pull_request:
+    paths:
+      - services/**
+
+jobs:
+  pre_ci:
+    name: Prepare CI environment
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Project
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+      - name: '[Push] Get commit message'
+        if: github.event_name == 'push'
+        id: push_get_commit_message
+        run: echo ::set-output name=push_commit_message::$(git log --format=%B -n 1 HEAD)
+      - name: '[Pull Request] Get commit message'
+        if: github.event_name == 'pull_request'
+        id: pr_get_commit_message
+        run: echo ::set-output name=pr_commit_message::$(git log --format=%B -n 1 HEAD^2)
+      - name: Add problem matchers
+        run: |
+          echo "::add-matcher::.github/problemMatchers/tsc.json"
+          echo "::add-matcher::.github/problemMatchers/eslint-stylish.json"
+    outputs:
+      commit_message: $( [ -z "${{ steps.pr_get_commit_message.outputs.pr_commit_message }}" ] && echo "${{ steps.push_get_commit_message.outputs.push_commit_message }}" || echo "${{ steps.pr_get_commit_message.outputs.pr_commit_message }}" )
+
+  BuildAndTest:
+    name: Build and test
+    runs-on: ubuntu-latest
+    if: "!contains(needs.pre_ci.outputs.commit_message, '[skip ci]')"
+    needs: pre_ci
+    steps:
+      - name: Checkout Project
+        uses: actions/checkout@v2
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '5.0.x'
+      - name: Restore CI Cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.nuget/packages
+          key: ${{ runner.os }}-nuget-${{ hashFiles('**/packages.lock.json') }}
```

Comment:
packages.lock.json? what's this for?

Reply:
https://docs.microsoft.com/en-gb/nuget/consume-packages/package-references-in-project-files#locking-dependencies uuhhh

Reply:
also https://github.com/actions/cache/blob/main/examples.md#c---nuget

Reply:
ah - we don't use lockfiles, so I don't think this is useful?